mgmt/src/models/external/overlay/vpcpeering.rs

```diff
@@ -53,11 +198,49 @@ impl VpcManifest {
             ..Default::default()
         }
     }
-    pub fn add_expose(&mut self, expose: VpcExpose) -> ConfigResult {
-        expose.validate()?;
+    fn validate_expose_addition(&self, expose: &VpcExpose) -> ConfigResult {
+        // Check that prefixes in the expose don't overlap with prefixes in other exposes
+        for other_expose in &self.exposes {
+            validate_overlapping(
+                &expose.ips,
+                &expose.nots,
+                &other_expose.ips,
+                &other_expose.nots,
+            )?;
+            validate_overlapping(
+                &expose.as_range,
+                &expose.not_as,
+                &other_expose.as_range,
+                &other_expose.not_as,
+            )?;
+        }
+        Ok(())
+    }
+    pub fn add_expose(&mut self, new_expose: VpcExpose) -> ConfigResult {
```

Comment:
I would not do validation in object builders but later (at least that's what we've done so far). The reason is that we may want to  dump the whole config when rejecting it. So, IMO, we should assemble the config, completely, and then validate it. This also reduces the scope of each method and the number of places where validations are done.

Reply:
Right, I added them “early” to avoid building a broken object, but you make a good point, I'll change this.